src/Broadway/EventStore/InMemoryEventStore.php

```diff
@@ -53,6 +56,7 @@ public function append($id, DomainEventStreamInterface $eventStream)
             $this->assertPlayhead($this->events[$id], $playhead);
 
             $this->events[$id][$playhead] = $event;
+            $this->allEvents[] = $event;
```

Comment:
I thought about this more and am realizing it would make it difficult to build a common test case for this if you cannot rely on different implementations returning the events in the same order.

I think that in general the idea here is sound, but I'm also thinking this could be problematic if you have a projector that relies on events from multiple aggregate root types. You may run into a case where you won't be able to rebuild the read model in the same way you expect.

I don't know if there is a way around this, though, so it may just have to be something that we have to accept. :)